wadas/domain/motion_detection.py

```diff
@@ -0,0 +1,61 @@
+"""Module containing logic to instantiate a Motion Detection thread to process video stream
+   integrated with Qt UI."""
+
+import time
+
+import cv2
+from PySide6.QtCore import QThread, Signal
+from PySide6.QtGui import QImage
+
+from wadas.domain.camera import Camera
+
+
+class MotionDetectionThread(QThread):
+    """Motion Detection QThread to integrate cv video stream into Qt Dialog"""
+
+    frame_ready = Signal(QImage)
+
+    def __init__(self, camera_index=0, parent=None):
+        super().__init__(parent)
+        self.running = False
+        self.camera_index = camera_index
+
+    def run(self):
+        self.running = True
+        cap = cv2.VideoCapture(self.camera_index)
+        background_sub = cv2.createBackgroundSubtractorMOG2()
+
+        if not cap.isOpened():
+            print("Error: Unable to open camera.")
+            return
+
+        while self.running:
+            ret, frame = cap.read()
+            if not ret:
+                break
+
+            # Apply background subtraction
+            foreground_mask = background_sub.apply(frame)
+            contours, _ = cv2.findContours(
+                foreground_mask, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE
+            )
+            for cnt in contours:
+                if cv2.contourArea(cnt) > Camera.detection_params["min_contour_area"]:
```

Comment:
Better to move Camera.detection_params["min_contour_area"] outside of the for loop, on a local var: it'll speed-up the execution.